.eslintrc

```diff
@@ -226,7 +226,7 @@
         "before": false
       }
     ],
-    "@typescript-eslint/semi": ["error"],
+    "@typescript-eslint/semi": ["error", "always"],
```

Comment:
LGTM, with one question. 

It seems that this arose because of an incorrect eslint configuration -- if we use a list then there should have been an additional element besides the initial "error". If on the other hand we just used "error" as a string then it would have worked too. (At least that's my understanding based on looking at https://eslint.org/docs/rules/semi .)

So, it's weird that this slipped through. Please could you check the eslint file to make sure that all the other rules are formatted correctly? (In fact, is there a way to "lint" the eslint file? It might be worth adding automation for that :P)

Reply:
There isn't a way to lint that file, it would just fail if the file was invalid in some way. That being said, I did an audit and it seems like the other rules are okay.

Reply:
^ that being said, the documentation that you saw was valid, but I think it was more of a function where it didn't pick this up and that we're using the [`typescript-eslint`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/semi.md) package instead of eslint directly here.

Reply:
So that's weird, I'm confused. Is typescript-eslint wrong (in terms of the README in the link you sent me)?

Reply:
I looked at the changelogs for ESlint and it looks like they had some bug fixes to their syntax trees in the later releases. That apparently also fixes the issues here.